bciers/libs/components/src/auth/SessionTimeoutHandler.tsx

```diff
@@ -21,20 +22,42 @@ const SessionTimeoutHandler: React.FC = () => {
   const [sessionTimeout, setSessionTimeout] = useState<number>(
     getExpirationTimeInSeconds(session?.expires),
   );
-  const [logoutUrl, setLogoutUrl] = useState<string | undefined>(undefined);
+
+  const logoutChannelRef = useRef<BroadcastChannel | null>(null);
 
   useEffect(() => {
-    // Fetch the logout URL from environment variables when component mounts
-    getEnvValue("SITEMINDER_KEYCLOAK_LOGOUT_URL")
-      .then(setLogoutUrl)
-      .catch((error) => console.error("Failed to fetch logout URL:", error));
+    const channel = new BroadcastChannel("logout");
+    logoutChannelRef.current = channel;
+
+    channel.onmessage = (event) => {
+      if (event === "logout") {
+        window.location.href = "/your-logout-page"; // or `logoutUrl`
+      }
+    };
+
+    return () => {
+      channel.close();
+    };
   }, []);
```

Comment:
This section is weirding me out because channel.onmessage doesn't seem to be doing anything, but I am properly being logged out and redirected to the logout page in both tabs. Nextauth uses broadcasting already (https://leapcell.medium.com/deep-dive-into-nextauth-js-a-powerful-and-flexible-authentication-solution-c9b4ebd37b42) so maybe it's some weird interaction between broadcasts?